x-pack/plugins/ml/public/shared_components/single_metric_viewer/single_metric_viewer.tsx

```diff
@@ -130,18 +139,24 @@ const SingleMetricViewerWrapper: FC<SingleMetricViewerPropsWithDeps> = ({
             ]);
             setSelectedJobWrapper({ job: jobs[0], stats: jobStats[0] });
           } catch (e) {
+            const error = extractErrorProperties(e);
+            // Could get 404 because job has been deleted
+            setErrorEncountered(true);
+
             if (onError) {
-              onError(new Error(errorMessage));
+              onError(
+                new Error(error.statusCode === 404 ? jobNotFoundErrorMessage : basicErrorMessage)
+              );
             }
           }
         }
       }
-      if (isMounted() === false) {
+      if (isMounted() === false || errorEncountered) {
         return;
       }
       fetchSelectedJob();
     },
-    [selectedJobId, mlApi, isMounted, onError]
+    [errorEncountered, selectedJobId, mlApi, isMounted, onError]
```

Comment:
This fixes the job fetch, but if I then hit the link to edit the job ID to one that does exist, and apply the changes in the flyout. the panel doesn't update to show the new job. It still shows the 'no known job' message.

Reply:
Good catch - fixed in 2539fc3820e98561c1599d7ea6418e63a0da742c